pkg/systemd/quadlet/quadlet.go

```diff
@@ -530,6 +518,34 @@ func ConvertContainer(container *parser.UnitFile, isUser bool) (*parser.UnitFile
 		podman.add("--secret", secret)
 	}
 
+	mounts := container.LookupAllArgs(ContainerGroup, KeyMount)
+	for _, mount := range mounts {
+		params := strings.Split(mount, ",")
+		paramsMap := make(map[string]string, len(params))
+		for _, param := range params {
+			kv := strings.Split(param, "=")
+			paramsMap[kv[0]] = kv[1]
+		}
+		if paramType, ok := paramsMap["type"]; ok {
+			if paramType == "volume" || paramType == "bind" {
+				if paramSource, ok := paramsMap["source"]; ok {
+					paramsMap["source"] = handleStorageSource(service, paramSource)
+				} else if paramSource, ok = paramsMap["src"]; ok {
+					paramsMap["src"] = handleStorageSource(service, paramSource)
+				}
+			}
+		}
+		paramsArray := make([]string, 0, len(params))
+		paramsArray = append(paramsArray, fmt.Sprintf("%s=%s", "type", paramsMap["type"]))
```

Comment:
Don't really understand this bit. Is it necessary to reconstruct the user-passed args? Podman should be able to interpret them regardless.

Reply:
When the type if `volume` and the `source` is a Quadlet based volume, the code changes the name of the volume. So, the name set in the .container file is not the same that will be used in the .service file. As a result, we cannot use the user provided string as is and have to reconstruct it.
I could change the code so it reconstructs the string only if changes were made, but I'm not sure it will produce better code.

Reply:
Ack, fair enough